### PR TITLE
Add api-request

### PR DESCRIPTION
**Pull request by apsavin**

### CODE REVIEW

**Comment by apsavin** [comment]
@jifeon for review please

File: blocks/api-request/api-request.browser.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/**@module api-request*/
+modules.define('api-request', ['inherit'], function (provide, inherit, ApiRequest) {
+    "use strict";
+
+    provide(inherit(ApiRequest, {
```
Comment (jifeon) [inline blocks/api-request/api-request.browser.js]: mm inherit, nice. Do not remember it was ealier..

File: blocks/api-request/api-request.browser.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+         * @param {Object} data
+         * @constructs
+         */
+        __constructor: function (data) {
+            this.__base();
+
+            /**
+             * @type {Object}
+             */
+            this.data = data;
```
Comment (jifeon) [inline blocks/api-request/api-request.browser.js]: why it's public, why you need it?

File: blocks/api-request/api-request.browser.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        },
+
+        /**
+         * @param {*} error
+         * @param {XMLHttpRequest} [xhr]
```
Comment (jifeon) [inline blocks/api-request/api-request.browser.js]: it's not an XMLHttRequest sometimes

**Pull request merged**